app/components/todos-route/component.js

```diff
@@ -12,41 +11,28 @@ export default Component.extend({
   store: service(),
 
   filtered: computed('todos.@each.isCompleted', 'filter', function() {
-    var filter = this.get('filter');
-    var all = this.get('todos');
-
-    if (filter === 'all') { return all; }
-
-    return all.filterBy('isCompleted', filter === 'completed');
+    switch(this.get('filter')) {
+    case 'active':
+      return this.get('active');
+    case 'completed':
+      return this.get('completed');
+    default:
+      return this.get('todos');
+    }
   }),
 
   completed: filterBy('todos', 'isCompleted', true),
   active: filterBy('todos', 'isCompleted', false),
+  allAreDone: computed.empty('active'),
 
-  inflection: computed('active.[]', function() {
-    var active = this.get('active.length');
+  inflection: computed('active.length', function() {
```

Comment:
another options is to use ember-inflector and one of the helpers, but I haven't been convinced that is important for such a simple example.